src/app/static/src/app/store/modelOptions/mutations.ts

```diff
@@ -1,12 +1,14 @@
 import {Mutation, MutationTree} from 'vuex';
 import {ModelOptionsState} from "./modelOptions";
 import {DynamicFormData, DynamicFormMeta} from "../../components/forms/types";
+import {PayloadWithType} from "../../types";
 
 type ModelOptionsMutation = Mutation<ModelOptionsState>
 
 export interface MetadataMutations {
     validate: ModelOptionsMutation,
-    update: ModelOptionsMutation
+    update: ModelOptionsMutation,
+    FormMetaUpdated: ModelOptionsMutation
```

Comment:
Should these Mutations all start with upper-case to be consistent with the other Mutations, or is there a reason why validate and update are lower-case?

Reply:
Yeah this is a weird one. `validate` and `update` get called as functions directly from the component via `mapMutations`, rather than committed from actions. 

We could fire off actions instead of mapping mutations directly into the component, or we could just stick to the TitleCase and have some weird function names in the component.

They also differ in signature - they take the payload directly rather than an action payload with type. I'm actually not clear on why we're usually passing an action with type and payload, and whether we should switch all mutations to this simpler signature.

Overall this is the bit I really don't like about Vuex - the matching of actions to mutations based on strings. I much prefer the Redux "reducer" pattern. Anyway, maybe we can chat about this next week and decide on a convention/pattern that we like.

Reply:
Hmm, in fact, in the docs they use `pascalCase` for all mutation names. But they also have a suggested pattern for mitigating the thing I don't like, the string matching, by using constants https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types :thinking: